### ISSUE TITLE
When rerunning full simulation the DB file does not get emptied first

### ISSUE THREAD

**Issue by KirstenVer** [APSIMInitiative/ApsimX#9263]
### What happened?

In changing the design of experiment - different N rates - and then rerunning the full lot, the old N rates remained in the DB file. Manual emptying resolves this, but could consider whether it's a fair expectation that it overwrites the old run in full?

### What Operating System are you on?

Windows

**Comment by par456** [comment]
So this goes back to a design choice around the datastore that was made a while back. If you are running an entire file, it makes sense to clear the entire datastore, but if you have a larger file with a lot of simulations in it, you might only want to run 1 simulation and not lose the data generated from running other simulations.

Maybe this could be made a bit smarter (say if you are running all simulations or clearing data from simulations that no longer exist) then it does a clear first, but we don't want to lose the functionality it current has.

Might get some input from others on this before we change this.

**Comment by sno036** [comment]
I agree that the functionality is important to retain. From my observations, users just get into a habit of clearing the DB (probably more frequently than they really need to).

**Comment by peter-devoil** [comment]
Users develop these habits as a result of painful experiences - finding that their db is full of rubbish that was deleted from the simulation ages ago.

**Comment by femiguez** [comment]
When I'm helping students I always clear the db first as they have run the simulations several times already with incorrect inputs. However, I'm not sure I'd change the current behavior, mostly because I'm used to it.

**Comment by lie112** [comment]
Maybe this should be an APSIM settings option. That way users needing the persistence of the database can get that, otherwise the default is to delete contents of database before each run which would certainly remove a lot of confusion for most users and avoid some niggly issues regarding plotting when old data is present, especially with experiments.

**Comment by ric394** [comment]
> Maybe this should be an APSIM settings option. That way users needing the persistence of the database can get that, otherwise the default is to delete contents of database before each run which would certainly remove a lot of confusion for most users and avoid some niggly issues regarding plotting when old data is present, especially with experiments.

I think this sounds like a good option
